### ISSUE TITLE
Using inexisting property as "discriminator" value

### ISSUE THREAD

**Issue by postatum** [aml-org/amf#478]
### Issue
Running code below should fail or report validation error because according to RAML 1.0 spec provided RAML input is invalid.

The spec says:
> discriminator?: ... The value must match the name of one of the `declared` properties of a type.

Spec link:
https://github.com/raml-org/raml-spec/blob/master/versions/raml-10/raml-10.md/#object-type

### AMF Version
4.0.0-SNAPSHOT

### Code
```sh
$ java -jar amf-4.0.0-SNAPSHOT.jar validate -in "RAML 1.0" -mime-in "application/yaml" -p "RAML" file://api.raml
{
  "@type": "http://www.w3.org/ns/shacl#ValidationReport",
  "http://www.w3.org/ns/shacl#conforms": true
}
```

### api.raml
```raml
#%RAML 1.0
title: My API With Types
types:
  Person:
    type: object
    discriminator: idontexist
    properties:
      name: string
      kind: string
  Employee:
    type: Person
    discriminatorValue: employee
    properties:
      employeeId: string
  User:
    type: Person
    discriminatorValue: user
    properties:
      userId: string

```

**Comment by BlockLucas** [comment]
https://www.mulesoft.org/jira/browse/APIMF-1533

**Comment by BlockLucas** [comment]
Resolved in 3.3.0
